src/main/java/com/epam/jmp/redislab/service/RateLimitRuleSearcher.java

```diff
@@ -0,0 +1,61 @@
+package com.epam.jmp.redislab.service;
+
+import com.epam.jmp.redislab.api.RequestDescriptor;
+import com.epam.jmp.redislab.configuration.ratelimit.RateLimitRule;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class RateLimitRuleSearcher {
+
+    private final List<RateLimitRule> rateLimitRules;
+
+    public RateLimitRuleSearcher(List<RateLimitRule> rateLimitRules) {
+        this.rateLimitRules = rateLimitRules;
+    }
+
+    public Optional<RateLimitRule> searchByRequestDescriptor(RequestDescriptor descriptor) {
+        return rateLimitRules.stream()
+                .filter(rule -> isDescriptorApplicableForRule(descriptor, rule))
+                .findFirst();
+    }
+
+    private boolean isDescriptorApplicableForRule(RequestDescriptor descriptor, RateLimitRule rule) {
```

Comment:
Quite complicated way to choose a rule. Somewhere is a mistake, see tests result:
![image](https://github.com/ShiniTada/rate-limit-business-service/assets/46779610/092bcf9a-f43b-490e-a3ea-bf0ca0805a9e)

Reply:
There are no any mistakes in rules.
The tests were failing because test configuration needed to tell RestTemplate to treat "429 Too Many Requests" as a normal response and not an error. I have added this configuration. Now all tests are passed. See: 
![image](https://github.com/ShiniTada/rate-limit-business-service/assets/41171038/360a4d75-0e30-48a0-b447-0eba2153793f)